### PR TITLE
Ziproxy 3.3.1 (new formula)

### PR DESCRIPTION
**Pull request by maxmouchet**
New formula for Ziproxy.

### CODE REVIEW

File: Library/Formula/ziproxy.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Ziproxy < Formula
+  homepage "http://ziproxy.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/ziproxy/ziproxy/ziproxy-3.3.1/ziproxy-3.3.1.tar.gz"
+  sha1 "7c8db842754d1e3e90b42c3c37f9b15e932a6545"
+
+  def install
+    ENV["LDFLAGS"] = "-lresolv"
```
Comment (mikemcquaid) [inline Library/Formula/ziproxy.rb]: Why is this needed?

File: Library/Formula/ziproxy.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--with-cfgfile=#{etc}/ziproxy/ziproxy.conf"
+
+    system "make", "install"
+
+    (etc/"ziproxy").install Dir["etc/ziproxy/*"]
```
Comment (mikemcquaid) [inline Library/Formula/ziproxy.rb]: Does `etc.install "ziproxy"` work?

File: Library/Formula/ziproxy.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  test do
+    system "ziproxy -d -p ziproxy.pid"
+
+    # Check ziproxy response headers
+    result = `curl -s -I http://127.0.0.1:8080`.include?('ziproxy')
```
Comment (mikemcquaid) [inline Library/Formula/ziproxy.rb]: You want to use `assert` and `shell_output` here.

**Comment by mikemcquaid** [comment]
Fails to install with what looks like missing dependencies:
http://bot.brew.sh/job/Homebrew%20Pull%20Requests/19588/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_ziproxy/

Also please fix `brew audit --strict ziproxy`: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/19588/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_ziproxy___strict/

File: Library/Formula/ziproxy.rb (hunk 1 above)
Comment (maxmouchet) [inline Library/Formula/ziproxy.rb]: It needs `_res_9_init`, which is part of libresolv.

File: Library/Formula/ziproxy.rb (hunk 2 above)
Comment (maxmouchet) [inline Library/Formula/ziproxy.rb]: `etc.install "etc/ziproxy"` do work indeed. Thanks !

File: Library/Formula/ziproxy.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/ziproxy.rb]: This should be supplied by the buildsystem. Can you please create and submit a patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/ziproxy.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                          "--prefix=#{prefix}",
+                          "--with-cfgfile=#{etc}/ziproxy/ziproxy.conf"
+
+    system "make", "install"
+
+    etc.install "etc/ziproxy" unless File.exist?("#{ etc }/ziproxy")
+  end
+
+  test do
+    system "ziproxy", "-d", "-p", "ziproxy.pid"
```
Comment (bfontaine) [inline Library/Formula/ziproxy.rb]: Please use `"#{bin}/ziproxy"` instead of `"ziproxy"`.

File: Library/Formula/ziproxy.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    # Check ziproxy response headers
+    assert shell_output("curl -s -I http://127.0.0.1:8080").include?("ziproxy")
+
+    system "ziproxy", "-k", "-p", "ziproxy.pid"
```
Comment (bfontaine) [inline Library/Formula/ziproxy.rb]: Same here.

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

File: Library/Formula/ziproxy.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  url "https://downloads.sourceforge.net/project/ziproxy/ziproxy/ziproxy-3.3.1/ziproxy-3.3.1.tar.gz"
+  sha1 "7c8db842754d1e3e90b42c3c37f9b15e932a6545"
+
+  depends_on "jasper"
+  depends_on "giflib"
+  depends_on "libpng"
+  depends_on "jpeg"
+
+  def install
+    ENV["LDFLAGS"] = "-lresolv"
```
Comment (mikemcquaid) [inline Library/Formula/ziproxy.rb]: Any idea why their buildsystem isn't passing this?

File: Library/Formula/ziproxy.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--with-cfgfile=#{etc}/ziproxy/ziproxy.conf"
+
+    system "make", "install"
+
+    etc.install "etc/ziproxy" unless File.exist?("#{ etc }/ziproxy")
```
Comment (mikemcquaid) [inline Library/Formula/ziproxy.rb]: Don't need the `unless`; this is handled automatically.

**Comment by mikemcquaid** [comment]
Fails to install on 10.8, 10.9 and needs audit fixes: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/27017/

**Comment by DomT4** [comment]
Ping.

**Comment by DomT4** [comment]
This seems to have gone dead. Closing consequently.

If you pick up interest later please feel free to resubmit as a new PR with the comments above addressed, Thanks!

Thank you for the submission!
